torch/csrc/jit/runtime/profiling_graph_executor_impl.cpp

```diff
@@ -222,6 +222,9 @@ bool guardDifferentiableGraph(Node* dnode) {
         }
       }
 
+      bool requires_grad = dni->type()->expectRef<TensorType>().requires_grad();
```

Comment:
this probably doesn't matter much, but `requires_grad` will return a `True` if requires_grad is `c10::nullopt` on the tensor type. `requiresGrad` preserves the optionality. We should probably use that api, just for clarity, even though later down the road if we don't know if a Value requires grad (c10::nullopt case) we will just assume that it does as you are doing here.